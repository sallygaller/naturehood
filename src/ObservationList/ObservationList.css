.ObservationList {
    display: flex;
    flex-direction: column;
    text-align: center;
    margin-left: 0px;
    padding: 20px;  
    font-family: 'Lato', sans-serif;
    border-style: solid;
    border-radius: 5px;
    border-image: url("../Utils/images/banner.jpg") 80 round;
    border-width: 6px;
}

.ObservationList h2 {
    font-family: 'Lato', sans-serif;
    text-align: center;
}

.ObservationList select {
    width: 25%;
}

.ObservationList ul, li {
    list-style-type: none;
    list-style-position: inside;
    padding: 0;
}

.ObservationList li {
    margin-left: 0px;
}

.ObservationList button {
    background-color: #012611;
    width: 50%;
    color: #f2f2f2;
    margin-top: 10px;
    padding: 5px;
    border-radius: 4px;
    cursor: pointer;
}

.ObservationList select {
    width: 50%;
}

@media screen and (min-width: 1000px) {
    .ObservationList button {
        width: 25%;
  }
}