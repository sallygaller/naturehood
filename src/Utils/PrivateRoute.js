import React from "react";
import { Route, Redirect } from "react-router-dom";
import TokenService from "../services/token-service";

export default function PrivateRoute({ component, ...props }) {
  const Component = component;
  return (
    <Route
      {...props}
      render={(componentProps) =>
        TokenService.hasAuthToken() ? (
          (console.log({ componentProps }, { props }),
          (<Component {...componentProps} {...props} />))
        ) : (
          <Redirect
            to={{
              pathname: "/login",
              state: { from: componentProps.location },
            }}
          />
        )
      }
    />
  );
}
