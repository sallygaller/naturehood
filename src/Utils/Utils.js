import moment from "moment";

export function dateFormat(observation) {
  const dateFormat = moment(observation.date).format("L");
  return dateFormat;
}

export function timeFormat(observation) {
  const timeFormat = moment(observation.time, "hh:mm:ss").format("hh:mm a");
  return timeFormat;
}

export const speciesTypes = [
  { id: 1, type: "Mammal" },
  { id: 2, type: "Fish" },
  { id: 3, type: "Bird" },
  { id: 4, type: "Arthropod" },
  { id: 5, type: "Amphibian" },
  { id: 6, type: "Reptile" },
];

export const testObservations = [
  {
    id: 1,
    species: "Robin",
    type: "Bird",
    date: "2020-12-25T08:00:00.000Z",
    time: "08:30:00",
    description: "I saw a robin at my feeder",
    lat: 45.593,
    lng: -122.755,
    date_added: "2020-12-25T09:28:32.615Z",
    neighbor: 1,
  },
  {
    id: 2,
    species: "Fox",
    type: "Mammal",
    date: "2020-12-15T08:00:00.000Z",
    time: "04:30:00",
    description: "I saw a mother fox and her cubs. The mother hissed at me.",
    lat: 47.593,
    lng: -122.755,
    date_added: "2020-12-15T16:28:32.615Z",
    neighbor: 1,
  },
  {
    id: 3,
    species: "Raccoon",
    type: "Mammal",
    date: "2020-12-05T08:00:00.000Z",
    time: "23:30:00",
    description: "A group of raccoons chased me down the street.",
    lat: 49.593,
    lng: -122.755,
    date_added: "2020-12-06T09:28:32.615Z",
    neighbor: 1,
  },
  {
    id: 4,
    species: "Skunk",
    type: "Mammal",
    date: "2020-12-08T08:00:00.000Z",
    time: "07:30:00",
    description: "I saw a skunk under my deck.",
    lat: 51.593,
    lng: -123.755,
    date_added: "2020-12-08T10:15:32.615Z",
    neighbor: 1,
  },
];
