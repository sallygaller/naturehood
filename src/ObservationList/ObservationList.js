import React from "react";
import { Link } from "react-router-dom";
import Observation from "../Observation/Observation";
import Analysis from "../Analysis/Analysis";
import { API_ENDPOINT } from "../config";
import "./ObservationList.css";

class ObservationList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      observations: [],
      error: null,
    };
  }

  setObservations = (observations) => {
    this.setState({
      observations: observations,
      error: null,
    });
    console.log(this.state.observations);
  };

  componentDidMount() {
    fetch(API_ENDPOINT)
      .then((res) => {
        if (!res.ok) {
          throw new Error(console.log(res.status));
        }
        return res.json();
      })
      .then(this.setObservations)
      .catch((error) => {
        console.error(error);
        this.setState({ error });
      });
  }

  render() {
    return (
      <div className="ObservationList">
        <h2>My Observations</h2>
        <div>
          <label htmlFor="Observations-sort">Sort by </label>
          <select>
            <option value="date-desc">Date (Newest to Oldest)</option>
            <option value="date-asc">Date (Oldest to Newest)</option>
            <option value="species">Species</option>
          </select>
        </div>
        <div>
          <ul>
            {this.state.observations.map((observation) => (
              <li key={observation.id}>
                <Observation observation={observation} />
              </li>
            ))}
          </ul>
          <Link to={"/add-observation"}>
            <button className="Observations-button" type="button">
              Add Observation
            </button>
          </Link>
        </div>
        <h3>Analysis</h3>
        <Analysis observations={this.state.observations} />
      </div>
    );
  }
}

export default ObservationList;
