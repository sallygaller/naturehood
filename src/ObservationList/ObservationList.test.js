import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import ObservationList from "./ObservationList";
import { observations } from "../Utils/TestFiles";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <ObservationList observations={observations} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <ObservationList name="ObservationList" observations={observations} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders the UI as expected with no observations", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <ObservationList name="ObservationList" observations={[]} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
