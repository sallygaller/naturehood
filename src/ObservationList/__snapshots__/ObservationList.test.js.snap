// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="ObservationList"
>
  <h2>
    My Observations
  </h2>
  <a
    href="/add-observation"
    onClick={[Function]}
  >
    <button
      className="ObservationList-button"
      type="button"
    >
      Add Observation
    </button>
  </a>
  <div>
    <ul />
  </div>
  <div
    className="Analysis"
  >
    <p>
      Total observations: 
      0
       
      <br />
      Most recent observation: 
      01/29/2021
    </p>
    <ul>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Mammal"
            className="SpeciesAnalysis-img"
            src="mammal.jpeg"
          />
          seen
        </div>
      </li>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Fish"
            className="SpeciesAnalysis-img"
            src="fish.jpeg"
          />
          seen
        </div>
      </li>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Bird"
            className="SpeciesAnalysis-img"
            src="bird.jpeg"
          />
          seen
        </div>
      </li>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Arthropod"
            className="SpeciesAnalysis-img"
            src="insect.jpeg"
          />
          seen
        </div>
      </li>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Amphibian"
            className="SpeciesAnalysis-img"
            src="amphibian.jpeg"
          />
          seen
        </div>
      </li>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Reptile"
            className="SpeciesAnalysis-img"
            src="reptile.jpeg"
          />
          seen
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders the UI as expected with no observations 1`] = `
<div
  className="ObservationList"
>
  <h2>
    My Observations
  </h2>
  <a
    href="/add-observation"
    onClick={[Function]}
  >
    <button
      className="ObservationList-button"
      type="button"
    >
      Add Observation
    </button>
  </a>
  <div>
    <ul />
  </div>
  <div
    className="Analysis"
  >
    <p>
      Total observations: 
      0
       
      <br />
      Most recent observation: 
      01/29/2021
    </p>
    <ul>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Mammal"
            className="SpeciesAnalysis-img"
            src="mammal.jpeg"
          />
          seen
        </div>
      </li>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Fish"
            className="SpeciesAnalysis-img"
            src="fish.jpeg"
          />
          seen
        </div>
      </li>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Bird"
            className="SpeciesAnalysis-img"
            src="bird.jpeg"
          />
          seen
        </div>
      </li>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Arthropod"
            className="SpeciesAnalysis-img"
            src="insect.jpeg"
          />
          seen
        </div>
      </li>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Amphibian"
            className="SpeciesAnalysis-img"
            src="amphibian.jpeg"
          />
          seen
        </div>
      </li>
      <li>
        <div
          className="SpeciesAnalysis"
        >
          0
           
          <img
            alt="Reptile"
            className="SpeciesAnalysis-img"
            src="reptile.jpeg"
          />
          seen
        </div>
      </li>
    </ul>
  </div>
</div>
`;
